app/src/main/java/com/nutomic/syncthingandroid/activities/WebGuiActivity.java

```diff
@@ -140,20 +140,31 @@ public void onCreate(Bundle savedInstanceState) {
         mWebView.getSettings().setDomStorageEnabled(true);
         mWebView.setWebViewClient(mWebViewClient);
         mWebView.clearCache(true);
+
+        // SyncthingService needs to be started from this activity as the user
+        // can directly launch this activity from the recent activity switcher.
+        startService(new Intent(this, SyncthingService.class));
     }
 
     @Override
     public void onServiceConnected(ComponentName componentName, IBinder iBinder) {
         super.onServiceConnected(componentName, iBinder);
-        getService().registerOnWebGuiAvailableListener(WebGuiActivity.this);
+        getService().registeronServiceStateChangeListener(this);
     }
 
     @Override
-    public void onWebGuiAvailable() {
-        if (mWebView.getUrl() == null) {
-            mWebView.stopLoading();
-            setWebViewProxy(mWebView.getContext().getApplicationContext(), "", 0, "localhost|0.0.0.0|127.*|[::1]");
-            mWebView.loadUrl(getService().getWebGuiUrl().toString());
+    public void onServiceStateChange(SyncthingService.State newState) {
```

Comment:
Fix for https://github.com/syncthing/syncthing-android/issues/1122 - Solution:
Only expose onWebGuiAvailable to RestApi (where the only important place from my point of view is where it's really needed) and implement onServiceStateChange() (before PR it's called onApiChange)